aqua-examples/ts-oracle/Readme.md

```diff
@@ -106,84 +161,140 @@ data Oracle:
 -- function collects the neighborhood peers
 -- retrieves and returns the timestamps 
-func ts_getter(node: string) -> []u64:
+-- the peer id gets from the wrapper of our service interface
+-- for handling timeouts from the neighborhood peers, the race pattern is used
+func ts_getter() -> []u64:
+  rtt = 1000 -- millis
+  msg = "timeout"
   res: *u64
-  on node:
-    k <- Op.string_to_b58(node)
+
+  services <- App.services()
+
+  on services.tsOracle.default!.peerId:
+    k <- Op.string_to_b58(services.tsOracle.default!.peerId)
     nodes <- Kademlia.neighborhood(k, nil, nil)
-    for n <- nodes par:
-      on n:
-        try:
-          res <- Peer.timestamp_ms()
-    Op2.identity(res!9)
+    if Op.array_length(nodes) > 0:
+      for n <- nodes par:
+        on n:
+          try:
+            res <- Peer.timestamp_ms()
+      -- whether we get res from all the nodes or rtt milliseconds passed, return res
+      join res[Op.array_length(nodes) - 1]
+      par Peer.timeout(rtt, msg)
+
   <- res
 
 -- oracle function operating on array of timestamps utilizing a distributed
 -- service to calculate the point_estimate
--- see src/main.rs for the service implementation
-func ts_oracle(node: string, oracle_service_id: string, min_points:u32) -> Oracle:
-  res <- ts_getter(node)
-  
-  on node:
-    TSOracle oracle_service_id
+-- see ts-oracle/modules/ts-oracle/src/main.rs for the service implementation
+-- the running service environment is injected using the generated
+-- wrapper defined in .fluence/aqua/deployed.app.aqua and
+-- imported in the beginning of the src/aqua/main.aqua
+func ts_oracle(min_points: u32) -> Oracle:
+  services <- App.services()
+  res <- ts_getter()
+  on services.tsOracle.default!.peerId:
+    TSOracle services.tsOracle.default!.serviceId
     oracle <- TSOracle.point_estimate(res, min_points)  -- calculate mode 
   <- oracle                                             -- and return to initiating peer
 ```
 
-We can run our Aqua `ts_oracle` script against the deployed processing service to get our oracle point estimate using `aqua run` (Note that you can replace the service id with the one you obtained from your deployment):
+We can run our Aqua `ts_oracle` function against the deployed processing service to get our oracle point estimate using `fluence run`:
 
 ```bash
-aqua run \
-     -i aqua \
-     -a /dns4/kras-02.fluence.dev/tcp/19001/wss/p2p/12D3KooWHLxVhUQyAuZe6AHMB29P7wkvTNMn7eDMcsqimJYLKREf \
-     -f 'ts_oracle("12D3KooWHLxVhUQyAuZe6AHMB29P7wkvTNMn7eDMcsqimJYLKREf", "ed657e45-0fe3-4d6c-b3a4-a2981b7cadb9", 5)'
+ fluence run \
+    -i src/aqua \
+    -f 'ts_oracle(5)'
 ```
 
-Which results in below but may be different for you:
+Please note that with `fluence run` we don't provide both the peer id and service id for our service. However, this information is used implicitly by the CLI tool, and is taken from definitions located in `.fluence/app.yaml` generated upon successful deployment.
```

Comment:
we don't provide  peer id and service id for our service
thee is no However; it is what it is. the information is fetched by the CLI tool from

Reply:
fixed